ext/js/language/language-transforms.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+
+/**
+ * @param {string} inflectedSuffix
+ * @param {string} deinflectedSuffix
+ * @param {string[]} conditionsIn
+ * @param {string[]} conditionsOut
+ * @returns {import('language-transformer').SuffixRule}
+ */
+export function suffixInflection(inflectedSuffix, deinflectedSuffix, conditionsIn, conditionsOut) {
+    const suffixRegExp = new RegExp(inflectedSuffix + '$');
+    return {
+        type: 'suffix',
+        isInflected: suffixRegExp,
+        deinflected: deinflectedSuffix,
+        uninflect: (text) => text.replace(suffixRegExp, deinflectedSuffix),
```

Comment:
I wonder if there is potentially a memory issue with doing the uninflect functions in this way. It would seem to be creating a redundant closure for every single invocation of `suffixInflection`, which may be suboptimal. I.e. `deinflectedSuffix` will be captured and stored in memory twice, once in the object that is being returned, and once in the implicit closure being created.

I'm comparing this to something like `uninflect: genericUninflect` where `genericUninflect` is a function that is passed a contextual object (basically what is currently the return value), or a class-like structure where the function can access the `this` members. This probably doesn't really matter though, just some food for thought.